exchanges/poloniex/poloniex_types.go

```diff
@@ -446,75 +125,906 @@ var WithdrawalFees = map[currency.Code]float64{
+	Amount        types.Number      `json:"amount"`
+	StopPrice     types.Number      `json:"stopPrice"`
+	CreateTime    types.Time        `json:"createTime"`
+	UpdateTime    types.Time        `json:"updateTime"`
+}
+
+// CancelOrderResponse represents a cancel order response instance.
+type CancelOrderResponse struct {
 	OrderID       string `json:"orderId"`
 	ClientOrderID string `json:"clientOrderId"`
 	State         string `json:"state"`
 	Code          int64  `json:"code"`
 	Message       string `json:"message"`
 }
 
-// WalletActivityResponse holds wallet activity info
-type WalletActivityResponse struct {
-	Deposits    []WalletDeposits    `json:"deposits"`
-	Withdrawals []WalletWithdrawals `json:"withdrawals"`
+// WsCancelOrderResponse represents a websocket cancel orders instance.
+type WsCancelOrderResponse struct {
+	OrderID int64 `json:"orderId"`
+	CancelOrderResponse
 }
 
-// WalletDeposits holds wallet deposit info
-type WalletDeposits struct {
-	DepositNumber int64         `json:"depositNumber"`
-	Currency      currency.Code `json:"currency"`
-	Address       string        `json:"address"`
-	Amount        float64       `json:"amount,string"`
-	Confirmations int64         `json:"confirmations"`
-	TransactionID string        `json:"txid"`
-	Timestamp     types.Time    `json:"timestamp"`
-	Status        string        `json:"status"`
+// CancelOrdersRequest represents cancel order request parameters
+type CancelOrdersRequest struct {
+	Symbol         currency.Pair `json:"symbol"`
+	OrderIDs       []string      `json:"ordIds,omitempty"`
+	ClientOrderIDs []string      `json:"clOrdIds,omitempty"`
 }
 
-// WalletWithdrawals holds wallet withdrawal info
-type WalletWithdrawals struct {
-	WithdrawalRequestsID int64         `json:"withdrawalRequestsId"`
-	Currency             currency.Code `json:"currency"`
-	Address              string        `json:"address"`
-	Amount               float64       `json:"amount,string"`
-	Fee                  float64       `json:"fee,string"`
-	Timestamp            types.Time    `json:"timestamp"`
-	Status               string        `json:"status"`
-	TransactionID        string        `json:"txid"`
-	IPAddress            string        `json:"ipAddress"`
-	PaymentID            string        `json:"paymentID"`
-}
-
-// TimeStampResponse returns the time
-type TimeStampResponse struct {
-	ServerTime types.Time `json:"serverTime"`
+// KillSwitchStatus represents a kill switch response
+type KillSwitchStatus struct {
+	StartTime        types.Time `json:"startTime"`
+	CancellationTime types.Time `json:"cancellationTime"`
+}
+
+// SmartOrderRequestRequest represents a smart trade order parameters
+type SmartOrderRequestRequest struct {
+	Symbol         currency.Pair `json:"symbol"`
+	Side           order.Side    `json:"side"`
+	TimeInForce    timeInForce   `json:"timeInForce,omitempty"`
+	AccountType    accountType   `json:"accountType,omitempty"`
+	Type           string        `json:"type,omitempty"`
```

Comment:
🚧 Change request:
This field was requested to be turned into a type as well.
It's been left out, but no explanation about why.
I'll include it in a patch